### PR TITLE
feat: cyclondx sbom custom property support

### LINKED ISSUE #4029: kubernetes bill of materials support
it is required to produce kubernetes (core components) bill of materials supported:
- kind
- minikube
- rancher
- openshift
- gke
- eks
- aks

### PR DESCRIPTION
**Pull request by chen-keinan**
## Description
cyclonedx custom property support

## Related issues
- Close #4029

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: pkg/sbom/cyclonedx/core/cyclonedx.go [hunk 1]
```diff
@@ -289,17 +295,23 @@ func (c *CycloneDX) Licenses(licenses []string) *cdx.Licenses {
...
-	})
+		cdxProps = append(cdxProps,
+			cdx.Property{
+				Name:  namespace + property.Name,
+				Value: property.Value,
+			})
+	}
 	sort.Slice(properties, func(i, j int) bool {
 		return properties[i].Name < properties[j].Name
 	})
```
Comment (knqyf263) [inline pkg/sbom/cyclonedx/core/cyclonedx.go:313]: You renamed `properties` to `cdxProps`. This sorting also needs to be updated.

```suggestion
	sort.Slice(cdxProps, func(i, j int) bool {
		return cdxProps[i].Name < cdxProps[j].Name
	})
```

File: pkg/sbom/cyclonedx/core/cyclonedx.go [hunk 2]
```diff
@@ -289,17 +295,23 @@ func (c *CycloneDX) Licenses(licenses []string) *cdx.Licenses {
 	return lo.ToPtr(cdx.Licenses(choices))
 }
 
-func (c *CycloneDX) Properties(props map[string]string) []cdx.Property {
-	properties := lo.MapToSlice(props, func(k, v string) cdx.Property {
-		return cdx.Property{
-			Name:  Namespace + k,
-			Value: v,
+func (c *CycloneDX) Properties(properties []Property) []cdx.Property {
+	cdxProps := make([]cdx.Property, 0)
```
Comment (knqyf263) [inline pkg/sbom/cyclonedx/core/cyclonedx.go:299]: nit
```suggestion
	cdxProps := make([]cdx.Property, len(properties))
```

or simply

```
var cdxProps []cdx.Property
```

File: pkg/sbom/cyclonedx/core/cyclonedx.go (hunk 1 above)
Comment (chen-keinan) [reply pkg/sbom/cyclonedx/core/cyclonedx.go:313]: thanks , good catch

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@chen-keinan I've merged another PR, and it's conflicting now. Can you please resolve it?

**Pull request merged**